src/misc.c

```diff
@@ -250,15 +256,16 @@ char *fgetll(char *s, size_t size, FILE *stream)
 		}
 	}
 }
-#endif
 
 void *strncpy_pad(void *dst, const void *src, size_t size, uint8_t pad)
 {
 	uint8_t *d = dst;
 	const uint8_t *s = src;
 
-	while (*s && size--)
-		*d++ = *s++;
+	if ( ((int)size) < 1) return dst;
+
+	while (*s && size)
+		*d++ = *s++, --size;
 	while (size--)
```

Comment:
By backing out this change, I will have to also comment out code to test this function.  The code (as written), will overwrite all of memory (i.e. ASAN catches this but gcore normally) if size is < strlen(dst).  

But I will back this out, and comment out this test (I have a couple others also), in unit-tests, making comments about why it is commented out in that source.